modules/dnn/include/opencv2/dnn/dnn.inl.hpp

```diff
@@ -199,6 +199,9 @@ inline void DictValue::release()
     case Param::REAL:
         delete pd;
         break;
+    default:
+        CV_Error_(Error::StsInternal, ("Unhandled type (%d)", static_cast<int>(type)));
+        break;
```

Comment:
Without "default:" branch and `switch` statement over "enum" - compilers / static analyzers are able to detect missing entries. It is useful when adding new entries to avoid missing of code refactoring.
Adding "default" blocks this detection, so this is non-wanted change (here and below). Build-time warning is preferable than runtime error.

Reply:
@alalek I thought you would mind if my patch exposes some hidden warnings, but if you do not mind, that is just perfect for me

Reply:
@alalek Done!

Reply:
@alalek A lot of warning [are there](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_android/builds/11851) due to this change.. is that really OK?

Reply:
Please use this pattern: https://github.com/alalek/opencv/commit/pr12310_enum_unhandled

Reply:
@alalek All done!